pkg/filters/health_check.go

```diff
@@ -22,10 +23,23 @@ func MaybeExecProbe(binary string, args string, execProbe []string) bool {
 	if err != nil {
 		return false
 	}
+
+	// Exec will append a script name to argument list if a sh/bash script is executed with a shebang,
+	// so we need to remove the first argument so that we can compare it to execProbe.
+	// e.g.
+	// "binary": "/health/ping_liveness_local.sh",
+	// "arguments": "/health/ping_liveness_local.sh 5"
+	// concatenated together will be ["/health/ping_liveness_local.sh", "/health/ping_liveness_local.sh", "5"],
+	// but execProbe will have only ["/health/ping_liveness_local.sh", "5"].
+	if binary == execProbe[0] && strings.HasSuffix(binary, ".sh") && len(argList) > 0 && binary == argList[0] {
+		argList = argList[1:]
+	}
```

Comment:
Hi thanks for the contributions.

the ".sh" suffix test is weak, as it is not a standard thing. Yes IIRC shebang execution will always end up with binary == argList[0].

So if the `execProbe []string` argument here is: `["/health/ping_liveness_local.sh", "5"]` , then you could write if test with something like:
```
if len(argList) > 0 && execProbe[0] == binary && argList[0] == binary && reflect.DeepEqual(argList[1:], execProbe) {
   
}
```

What do you think?

Reply:
Hi, thanks for comments, I agree that sh prefix is a bit redundant, that was just a double-protection from accidentally filtering scripts that are for some reason run with their name as the first argument.

In any case I added your proposal, except I don't do `reflect.DeepEqual(argList[1:], execProbe)` because comparison is done below.

What do you think?

Reply:
Sounds good alright!